Apache Ranger plugin (#13297)
trinodb/trino: docs/src/main/sphinx/connector/ranger.rst

> =======================

Reviewer: It is not a connector but system access control

Reviewer 2: I agree, this should likely live under [/docs/src/main/sphinx/security](https://github.com/trinodb/trino/tree/master/docs/src/main/sphinx/security) in a file called `ranger-access-control.rst`.

Apache Ranger connector
=======================

.. raw:: html

>   <img src="../_static/img/apache_ranger.png" class="connector-logo">

Reviewer 2: We should use [the latest Ranger logo](https://ranger.apache.org/ranger.jpg). We all love the elephant but anything I see with it now I associate with "outdated and slow".

https://ranger.apache.org/

> Apache Ranger is a framework to enable comprehensive data security across many platforms. Trino is one of these platforms.

Reviewer 2: ```suggestion
Apache Ranger is a framework to enable comprehensive data security across many platforms, including Trino.
```

Requirements
------------

To connect to Apache Ranger you need:

> * Apache Ranger installed, up and running. Compatible with Ranger v2.1.x, 2.2.x, 2.3.x

Reviewer: How do we verify that?

Reviewer 2: @dprophet we should add some tests for this. @kokosing recommends something like [TestSqlStandardAccessControlChecks](https://github.com/trinodb/trino/blob/403/testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestSqlStandardAccessControlChecks.java) and we need some environemnt with additional services, so we could follow [EnvMultinodeKafka](https://github.com/trinodb/trino/blob/403/testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/environment/EnvMultinodeKafka.java).

is a requirement if you are going to use corporate AD/LDAP to manage access controls.

service_name as defined in the Ranger UI

hadoop_config (aka trino-ranger-site.xml): is the Ranger site file required to connect
to corporate AD/LDAP systems. When the user logs into trino, this is the file used to
connect to the AD/LDAP system and get a list of the users groups.

audit_resource: Ranger can be configured to send reports to various systems.
At the moment this file is required for legacy reasons and should be cleaned up
in the figure. all DEFAULT settings is recommented. If you want auditing use the
trino-http-event-listener and post those events to the HTTP service of your choice.
We recommend using it to post to Kafla but its very flexible.

security_resource (aka trino-ranger-security.xml): Configures the connectivity
between the trino-ranger plugin and the Apache Ranger running service.

policy_manager_ssl_resource (aka trino-ranger-policymgr-ssl.xml): Used to setup
up 2 way SSL client/server validation

Full Enterprise ready Apache Ranger setup
-----------------------------------------

Install and startup Ranger:

> * Quick and dirty setup for Ranger

Reviewer 2: See README comments on how to break this up and update terminology.
